Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py

```diff
@@ -0,0 +1,193 @@
+
+            file_data = base64.b64decode(payload) if attachment.get('binary') else payload
+
+            # save the attachment
+            file_result = fileResult(attachment.get('filename'), file_data, attachment.get('mail_content_type'))
+
+            # check for error
+            if file_result['Type'] == entryTypes['error']:
+                demisto.error(file_result['Contents'])
+                raise Exception(file_result['Contents'])
+
+            file_names.append({
+                'path': file_result['FileID'],
+                'name': attachment.get('filename')
+            })
+        return file_names
+
+    def convert_to_incident(self) -> Dict[str, Any]:
+        """
+        Convert an Email class instance to a demisto incident
+        Returns:
+            A dict with all relevant fields for an incident
+        """
+        return {
+            'occurred': self.date.isoformat(),
+            'created': datetime.now(timezone.utc).isoformat(),
+            'details': self.text or self.html,
+            'name': self.subject,
+            'attachment': self.parse_attachment(),
+            'rawJSON': json.dumps(self.raw_json)
+        }
+
+    def generate_raw_json(self):
+        raw_json = {
+            'to': self.to,
+            'cc': self.cc,
+            'from': self.from_,
+            'format': self.format,
+            'text': self.text,
+            'subject': self.subject,
+            'attachments': ','.join([attachment['filename'] for attachment in self.attachments]),
+            'rawHeaders': self.headers
+        }
+        if self.html:
+            raw_json['html'] = self.html
+        if self.raw_body:
+            raw_json['rawBody'] = self.raw_body
+        return raw_json
+
+
+def fetch_incidents(client,
+                    last_run,
+                    first_fetch_time):
+    """
+    This function will execute each interval (default is 1 minute).
+
+    Args:
+        client (Client): HelloWorld client
+        last_run (dateparser.time): The greatest incident created_time we fetched from last fetch
+        first_fetch_time (dateparser.time): If last_run is None then fetch all incidents since first_fetch_time
+
+    Returns:
+        next_run: This will be last_run in the next fetch-incidents
+        incidents: Incidents that will be created in Demisto
+    """
+    # Get the last fetch time, if exists
+    last_fetch = last_run.get('last_fetch')
+
+    # Handle first time fetch
+    if last_fetch is None:
+        latest_created_time = dateparser.parse(f'{first_fetch_time} UTC')
+    else:
+        latest_created_time = datetime.fromisoformat(last_fetch)
+    messages_query = generate_search_query(latest_created_time, PERMITTED_FROM_ADDRESSES, PERMITTED_FROM_DOMAINS)
+    messages = client.search(messages_query)
+    messages = messages[:50]
+    mails_fetched = []
+    for uid, message_data in client.fetch(messages, 'RFC822').items():
+        message_bytes = message_data.get(b'RFC822')
```

Comment:
why are we working with bytes here? what other options we got?

Reply:
None the keys we have with message_data are `b'SEQ` which is a number and the bytes data.
Why is it matter?

Reply:
just wondering.. if that's the only option then ok